lib/widgets/new_dm_sheet.dart

```diff
@@ -0,0 +1,327 @@
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox(
+      height: MediaQuery.of(context).size.height * 0.95,
+      child: Column(children: [
+        NewDmHeader(
+          selectedUserIds: selectedUserIds,
+          onClose: () => Navigator.pop(context)),
+        NewDmSearchBar(
+          searchController: searchController,
+          scrollController: scrollController,
+          selectedUserIds: selectedUserIds),
+        Expanded(child: NewDmUserList(
+          filteredUsers: filteredUsers,
+          selectedUserIds: selectedUserIds,
+          onUserSelected: (userId) {
+            setState(() {
+              if (selectedUserIds.contains(userId)) {
+                selectedUserIds.remove(userId);
+              } else {
+                selectedUserIds.add(userId);
+              }
+              scrollToSearchField();
+            });
+          }))]));
+  }
+}
+
+class NewDmHeader extends StatelessWidget {
+  const NewDmHeader({
+    super.key,
+    required this.selectedUserIds,
+    required this.onClose});
+
+  final Set<int> selectedUserIds;
+  final VoidCallback onClose;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final store = PerAccountStoreWidget.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    return Container(
+      padding: const EdgeInsets.all(8),
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+        children: [
+          _buildBackButton(context, designVariables, zulipLocalizations),
+          Text(zulipLocalizations.newDmSheetScreenTitle,
+            style: TextStyle(color: designVariables.title,fontSize: 20, fontWeight: FontWeight.w600)),
+          _buildNextButton(context, store, designVariables, zulipLocalizations)]));
+  }
+
+  Widget _buildBackButton(BuildContext context, DesignVariables designVariables,
+      ZulipLocalizations zulipLocalizations) => TextButton(
+    onPressed: onClose,
+    child: Row(
+      mainAxisSize: MainAxisSize.min,
+      children: [
+        Icon(ZulipIcons.chevron_left, size: 24, color: designVariables.icon),
+        Text(zulipLocalizations.newDmSheetBackButtonLabel,
+          style: TextStyle(fontSize: 20, color: designVariables.icon))]));
+
+  Widget _buildNextButton(BuildContext context, PerAccountStore store,
+      DesignVariables designVariables, ZulipLocalizations zulipLocalizations) => TextButton(
+    onPressed: selectedUserIds.isEmpty ? null : () {
+      final narrow = DmNarrow.withOtherUsers(
+        selectedUserIds,
+        selfUserId: store.selfUserId);
+      Navigator.pop(context);
+      Navigator.push(context, MessageListPage.buildRoute(
+        context: context,
+        narrow: narrow));
+    },
+    child: Row(
+      mainAxisSize: MainAxisSize.min,
+      children: [
+        Text(zulipLocalizations.newDmSheetNextButtonLabel,
+          style: TextStyle(
+            fontSize: 20,
```

Comment:
Similar comment on line-height here. Be sure to check other `TextStyle`'s as well.